dbt/include/snowflake/macros/catalog.sql

```diff
@@ -70,3 +62,36 @@
   {{ return(run_query(query)) }}
 
 {%- endmacro %}
+
+
+{% snowflake__get_catalog_where_clause(schemas, relations) %}
+
+    {% if schemas is not none %}
+        where (
+            {%- for schema in schemas -%}
+                upper("table_schema") = upper('{{ schema }}')
+                {%- if not loop.last %} or {% endif -%}
+            {%- endfor -%}
+        )
+
+    {% elif relations is not none %}
```

Comment:
This is the new where clause that can take a set or BaseRelation instances. It assumes that `schema` is defined on all of them and `identifier` is defined on the instances that represent tables. This has the effect of allowing a mixture of tables and sets, if we want to support that. I condition based on the existence of `identifier`, which is a bit loose. There is no `is_schema` property on `BaseRelation` and I don't know that we should add one since `Schema` is not in `RelationType`; but it's another option.